21-Hello-World/05-Application-Structure/src/02-MTL/02-Implementing-a-Monad-Transformer/03-A-Magical-Monad.md

```diff
@@ -117,14 +115,34 @@ instance functor :: (Monad monad) => Functor (StateT state monad) where
       -> StateT state monad a
       -> StateT state monad b
   map f (StateT g) = StateT (\state ->
-      let
-        (Tuple value state2) = g state
-        b = f value
-      in
-        pure (Tuple b state2)
+      (g state) >>= (\(Tuple value state2) ->
+        let b = f value in
+        pure $ Tuple b state2
+      )
     )
+
+-- Finally, we can using `do notation` in this case
+instance functor :: (Monad monad) => Functor (StateT state monad) where
+  map :: forall a b
+       . (a -> b)
+      -> StateT state monad a
+      -> StateT state monad b
+  map f (StateT g) = StateT \state ->
+    do
+      (Tuple value state2) <- g state
+      pure $ Tuple (f value) state2
```

Comment:
I think I'd prefer this to show up at the very end. My goal here is not to be "readable" per say as much as it is to break down how the changes occur step by step. The FAABM section at the end shows the more readable version.